### PR TITLE
Docsp-35796: Add info about new Sync Schema Migrations to SDK pages

### PR DESCRIPTION
**Pull request by MongoCaleb**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-35796

- [Fluttter](https://preview-mongodbmongocaleb.gatsbyjs.io/realm/DOCSP-35796/sdk/flutter/realm-database/model-data/update-realm-object-schema/)
- [Java](https://preview-mongodbmongocaleb.gatsbyjs.io/realm/DOCSP-35796/sdk/java/model-data/modify-an-object-schema/)
- [Kotlin](https://preview-mongodbmongocaleb.gatsbyjs.io/realm/DOCSP-35796/sdk/kotlin/realm-database/schemas/change-an-object-model/)
- [.NET](https://preview-mongodbmongocaleb.gatsbyjs.io/realm/DOCSP-35796/sdk/dotnet/model-data/change-an-object-model/)
- [NodeJS](https://preview-mongodbmongocaleb.gatsbyjs.io/realm/DOCSP-35796/sdk/node/model-data/modify-an-object-schema/)
- [React Native](https://preview-mongodbmongocaleb.gatsbyjs.io/realm/DOCSP-35796/sdk/react-native/model-data/change-an-object-model/)
- [Swift](https://preview-mongodbmongocaleb.gatsbyjs.io/realm/DOCSP-35796/sdk/swift/model-data/change-an-object-model/)

### CODE REVIEW

File: source/includes/baas-server-migration-note.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+- Support server-side versioning. Device SDKs do not currently support client-side 
+  schema versioning.
+- Automatically migrate objects to the latest server-side schema version without 
+  requiring a manual migration.
+
+When working with a synced realm, you can update your schema in Atlas App Services 
+and Device Sync will update the local realm file automatically the next time it 
+connects . For more information on making server-side schema changes and handling 
+the differences between the server-side schema and your client data model, refer 
+to the Update a Data Model page in the App Services documentation.
```
Comment (cbullinger) [inline source/includes/baas-server-migration-note.rst:22]:
```suggestion
to the :ref:`Update a Data Model <synced-schema-overview>` page in the App Services documentation.
```

File: source/includes/baas-server-migration-note.rst [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+- Support making non-breaking and breaking changes.
+- Support server-side versioning. Device SDKs do not currently support client-side 
+  schema versioning.
+- Automatically migrate objects to the latest server-side schema version without 
+  requiring a manual migration.
+
+When working with a synced realm, you can update your schema in Atlas App Services 
+and Device Sync will update the local realm file automatically the next time it 
+connects . For more information on making server-side schema changes and handling
```
Comment (cbullinger) [inline source/includes/baas-server-migration-note.rst:20]:
```suggestion
connects. For more information on making server-side schema changes and handling 
```

File: source/includes/baas-server-migration-note.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+an unsynced (Iocal-only) realm.
+
+Update a Data Model in a Synced Realm
+-------------------------------------
+If your app uses Atlas Device Sync to synchronize data with MongoDB Atlas, and 
+you are working with a synced realm, schema changes are handled differently.
+
+Synced realms:
+
+- Support making non-breaking and breaking changes.
```
Comment (cbullinger) [inline source/includes/baas-server-migration-note.rst:12]:
```suggestion
- Support making non-breaking and breaking changes. For more information, refer to :ref:`Breaking Changes <destructive-changes-synced-schema>` in the App Services documentation.
```

File: source/includes/baas-server-migration-note.rst [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+- Support server-side versioning. Device SDKs do not currently support client-side 
+  schema versioning.
```
Comment (cbullinger) [inline source/includes/baas-server-migration-note.rst:14]:
```suggestion
  schema versioning in synced realms.
```

File: source/includes/tip-bypass-migration.rst [hunk 5]
```diff
@@ -0,0 +1,9 @@
+.. tip:: Bypass Migration During Development
+
+   When **developing or debugging** your application, you may prefer to delete the
+   realm instead of migrating it. Use the
+   :realm-react-sdk:`BaseConfiguration.deleteRealmIfMigrationNeeded
+   <types/Realm.BaseConfiguration.html>` property to delete the database automatically
+   when a schema mismatch requires a migration. 
+
+   **Never release an app to production with this property set to ``true``.**
```
Comment (cbullinger) [inline source/includes/tip-bypass-migration.rst:9]:
```suggestion
   Ensure that this property is always set to ``false`` for production apps.
   **Never release an app to production with this property set to true.**
```
some wonky formatting here; the code backticks don't play well with the bold, I guess?

File: source/includes/baas-server-migration-note.rst [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+- Automatically migrate objects to the latest server-side schema version without 
+  requiring a manual migration.
+
+When working with a synced realm, you can update your schema in Atlas App Services
```
Comment (cbullinger) [inline source/includes/baas-server-migration-note.rst:18]: Suggest we bump this one sentence to go before the "Synced realms:" list

**Review by cbullinger — APPROVED** [review]
lgtm! Thanks so much for your help on this! The only other suggestion would be to increase the `On this page` depth by one, so the updated subheadings are included in the toc, but that's not blocking at all

**Comment by dacharyc** [comment]
This has become a "won't do" so closing this PR.

**Pull request closed without merging**